jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/commands/ExecuteDeadlinesCommand.java

```diff
@@ -171,9 +171,13 @@ public Void execute(Context context) {
 	        }
 	        
 	        deadline.setEscalated(true);
-	        persistenceContext.updateDeadline(deadline);
-	        persistenceContext.updateTask(task);
-            DeadlineSchedulerHelper.rescheduleDeadlinesForTask((InternalTask) task, ctx, type);
+			persistenceContext.updateDeadline(deadline);
+			persistenceContext.updateTask(task);
+
+	        if(deadline.getRepeatCount() != 0) {
```

Comment:
where is the logic regarding the number of repetition ? I mean you set the variable in 
https://github.com/kiegroup/jbpm/pull/1493/files#diff-1a2981938587c542131221bf215dea1cR215

but don't see where you are decrementing till reach zero

Reply:
@elguardian need some advice here - rescheduling always re-parses all task deadlines from scratch, in which case the repeat count even if decremented will always go back to for example 3 in "R3/...". Since you wrote the rescheduling, is is possible to get it to with using existing task deadlines rather than reparsing each time ?

Reply:
nvm :) instead of re-parsing each time we now generate all the dealines up-front for repeat count expires